src/compiler/crystal/loader/unix.cr

```diff
@@ -20,6 +20,21 @@
 class Crystal::Loader
   alias Handle = Void*
 
+  class LoadError
+    def self.new_dl_error(message)
+      if char_pointer = LibC.dlerror
+        new(String.build do |io|
+          io << message
+          io << " ("
+          io.write_string(Slice.new(char_pointer, LibC.strlen(char_pointer)))
+          io << ")"
+        end)
```

Comment:
Using `String.build` here to avoid two intermediary string allocations. Maybe that's overkill for error handling...